### PR TITLE
Fix: break file-download for prevent ErrnoException

### PR DESCRIPTION
**Pull request by alvince**
Prevent android.system.ErrnoException: write failed: ENOSPC (No space left on device) from OutputResourceWrapper.write()

### CODE REVIEW

File: fetch2/src/main/java/com/tonyodev/fetch2/downloader/FileDownloader.kt [hunk 1]
```diff
@@ -33,4 +35,40 @@ interface FileDownloader : Runnable {
...
+            true
+        }
+    }
+
+}
+
+/**
+ * Reserved storage space for operation of the system, _128Mb_
+ */
+internal const val SPACE_RESERVED_DOWNLOAD = 134217728L
```
Comment (AlexStarc) [inline fetch2/src/main/java/com/tonyodev/fetch2/downloader/FileDownloader.kt]: Is it possible to make it configurable?

Comment (alvince) [reply fetch2/src/main/java/com/tonyodev/fetch2/downloader/FileDownloader.kt]: yep, it should change to configurable

**Comment by tonyofrancis** [comment]
@alvince looks like we have some merge conflicts.

**Comment by alvince** [comment]
> @alvince looks like we have some merge conflicts.

I'm sorry that I didn't handle it. I am so busy recently
Commits rebased, please check it @tonyofrancis

**Review by AlexStarc — APPROVED** [review]

**Comment by tonyofrancis** [comment]
@alvince once again I broke your PR.

**Pull request closed without merging**